### PR TITLE
Fix live doc build by saving runner space

### PR DESCRIPTION
**Pull request by kwankyu**
as seen in: https://github.com/sagemath/sage/actions/runs/9333677258/job/25690941738

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -168,12 +168,20 @@ jobs:
...
         with:
           name: docs
           path: docs.zip
 
+      - name: Remove docs
+        if: (success() || failure()) && steps.upload.outcome == 'success'
+        run: |
+          set -ex
+          # We remove previous docs directory to save space on runner device
+          rm -rf docs/
```
Comment (mkoeppe) [inline .github/workflows/doc-build.yml:184]: Simpler to do that right after line 168

Comment (kwankyu) [reply .github/workflows/doc-build.yml:184]: The "Upload docs" needs `docs/` directory and its contents. No?

Comment (mkoeppe) [reply .github/workflows/doc-build.yml:184]: It's uploading the zip file

Comment (kwankyu) [reply .github/workflows/doc-build.yml:184]: Ah, the zip file is not in the directory. 

Then perhaps removing the zip file as well would help. What do you think?

File: .github/workflows/doc-build.yml [hunk 2]
```diff
@@ -195,11 +203,11 @@ jobs:
...
           mkdir -p ./livedoc
-          cp -r -L /sage/local/share/doc/sage/html ./livedoc
-          cp -r -L /sage/local/share/doc/sage/pdf ./livedoc
-          cp  /sage/local/share/doc/sage/index.html ./livedoc
+          # We copy everything to a local folder
+          docker cp BUILD:/sage/local/share/doc/sage/html livedoc
+          docker cp BUILD:/sage/local/share/doc/sage/pdf livedoc
+          docker cp BUILD:/sage/local/share/doc/sage/index.html livedoc
           zip -r livedoc.zip livedoc
-        shell: sh .ci/docker-exec-script.sh BUILD . {0}
```
Comment (kwankyu) [inline .github/workflows/doc-build.yml:213]: Am I doing things right here?

File: .github/workflows/doc-build.yml (hunk 1 above)
Comment (kwankyu) [reply .github/workflows/doc-build.yml:184]: We remove `docs.zip` file too.

File: .github/workflows/doc-build.yml (hunk 2 above)
Comment (kwankyu) [reply .github/workflows/doc-build.yml:213]: I am asking because there is `-L` previously, but `docker cp` does not have the flag.

Comment (mkoeppe) [reply .github/workflows/doc-build.yml:213]: For the corresponding option for `docker cp`, see https://github.com/sagemath/sage/pull/36730/files

Comment (mkoeppe) [reply .github/workflows/doc-build.yml:213]: But the right way to do this, I think, is to finally do https://github.com/sagemath/sage/issues/35538#issuecomment-1532477914 as part of the docbuild...

Comment (kwankyu) [reply .github/workflows/doc-build.yml:213]: Thanks. I use `-L` equivalent to `--follow-link` for now.

Comment (kwankyu) [reply .github/workflows/doc-build.yml:213]: Rather more explicit `--follow-link` as in #36730.

**Comment by mkoeppe** [comment]
Is there a test run already somewhere?

**Comment by kwankyu** [comment]
I ran it here: https://github.com/kwankyu/sage/actions/runs/9424131740 Let's see.

**Comment by kwankyu** [comment]
It works well.

**Comment by mkoeppe** [comment]
Then let's merge it.

**Comment by kwankyu** [comment]
Thanks!

**Comment by kwankyu** [comment]
I am confused by this: https://github.com/sagemath/sage/actions/runs/9357979101

That run succeeded in building live doc. Then this PR is not really needed...?

**Comment by kwankyu** [comment]
In particular, this step
```
      - name: Copy live doc
        id: copylivedoc
        if: (success() || failure()) && steps.buildlivedoc.outcome == 'success'
        run: |
          mkdir -p ./livedoc
          cp -r -L /sage/local/share/doc/sage/html ./livedoc
          cp -r -L /sage/local/share/doc/sage/pdf ./livedoc
          cp  /sage/local/share/doc/sage/index.html ./livedoc
          zip -r livedoc.zip livedoc
        shell: sh .ci/docker-exec-script.sh BUILD . {0}
```
worked as it is in the run. I thought this step would not work correctly because the commands run inside the docker container and hence create the file `livedoc.zip` inside the container, and the upload step fails because the zip file is not accessible. I am misunderstanding something. What is it?

**Comment by mkoeppe** [comment]
The container mounts the current directory (`.`) from the worker. These steps directly write to the worker file system.

**Comment by kwankyu** [comment]
I see. That explains it. 

How about the space problem? Should we see the success of the last run as a sheer luck?

**Comment by mkoeppe** [comment]
It's better to save the space. It's clear that we are operating very close to the edge with our builds on these runners

**Comment by kwankyu** [comment]
I reviewed the last runs for from beta5 to beta8. Only the run beta8 failed by lack of space. 

> It's better to save the space. It's clear that we are operating very close to the edge with our builds on these runners

OK. Thanks.

**Pull request merged**